### PR TITLE
[Imported] Add some info make docs related deployment more clarify

### PR DESCRIPTION
**Pull request by mrtuvn**
## Purpose of this pull request

This pull request (PR) add some info make docs related pipeline build deployment more clarify

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.4/config-guide/deployment/pipeline/technical-details.html
https://devdocs.magento.com/guides/v2.4/config-guide/deployment/pipeline/build-system.html
https://devdocs.magento.com/guides/v2.4/config-guide/deployment/pipeline/example/shared-configuration.html
https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/cache_for_frontdevs.html
https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/themes/theme-create.html
https://devdocs.magento.com/guides/v2.4/howdoi/php/php_clear-dirs.html

## Links to Magento source code

-  ...

CC: @hostep @Eddcapone for review
CC: @sharkySharks @paliarush @drewML @dmytro-ch may  have knowledge of how build servers 
@sidolov i think we need someone have experiences with build system(cloud or normal server)  for confirm this

Should we need database connection in build system ? @kandy 
https://devdocs.magento.com/guides/v2.4/config-guide/deployment/pipeline/build-system.html <== reference this

### CODE REVIEW

**Comment by hostep** [comment]
This needs to be reviewed by somebody with extensive knowledge of how build servers work. I don't, but the change where it removes " Magento database connection" is definitely incorrect. See https://github.com/magento/devdocs/issues/8690#issuecomment-780350440

**Comment by mrtuvn** [comment]
Agreed!

File: src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md [hunk 1]
```diff
@@ -136,9 +149,14 @@ On your production system:
 1. If you installed components that changed the database schema, run `magento setup:upgrade --keep-generated` to update the database schema and data, preserving generated static files.
 1. To set system-specific settings, use either the `magento config:set` command or environment variables.
 1. To set sensitive settings, use either the `magento config:sensitive:set` command or environment variables.
+1. Use the `magento setup:di:compile` command to compile code.
+1. Use the `magento setup:static-content:deploy` command to update static file view files.
```
Comment (hostep) [inline src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md:153]: These two steps already happened on the build server, so they shouldn't be re-run again here, right? The idea of this way of working is that this runs fast with as little downtime as possible, adding these two command here will defeat this purpose.

Comment (mrtuvn) [reply src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md]: Updated

File: src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md [hunk 2]
```diff
@@ -112,6 +112,19 @@ On your development system:
 
 1. Add your modified code and the shared configuration to source control.
 
+1. For clear generated php code, static assets files while in development developers can use commands below.
+
+   ```bash
+      rm -rf var/view_preprocessed/*
+      rm -rf pub/static/frontend/* pub/static/adminhtml/*
```
Comment (kandy) [inline src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md:119]:
```suggestion
      rm -rf pub/static/*/* 
```

File: src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md [hunk 3]
```diff
@@ -112,6 +112,19 @@ On your development system:
...
+      rm -rf generated/code/*
```
Comment (kandy) [inline src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md:120]:
```suggestion
      rm -rf generated/*/*
```

**Comment by kandy** [comment]
> Should we need database connection in build system ?

Only in case if all needed information is dumped to config.php (see https://devdocs.magento.com/guides/v2.4/config-guide/cli/config-cli-subcommands-config-mgmt-export.html). Otherwise, connection to DB is required

File: src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md (hunk 2 above)
Comment (hostep) [reply src/guides/v2.3/config-guide/deployment/pipeline/technical-details.md:119]: And also remove the `-f` flag. It's bad practice to teach people to use this flag, it should only be used in very exceptional cases.

**Comment by dobooth** [comment]
@magento import code to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth the branch with code successfully imported into`magento-devdocs/devdocs` repository. Branch name: `imported-magento-devdocs-8697`.

**Pull request merged**